src/_repobee/command/repos.py

```diff
@@ -441,17 +463,44 @@ def _set_pull_ff_only(local_repos: List[plug.StudentRepo]) -> None:
         git.set_gitconfig_options(repo.path, {"pull.ff": "only"})
 
 
+def _with_output_paths(
+    repos: Iterable[plug.StudentRepo], directory_layout: DirectoryLayout
+) -> List[plug.StudentRepo]:
+    base_dir = pathlib.Path(".").resolve()
+    repo_path_func = _select_repo_path_function(directory_layout)
+    return [repo.with_path(repo_path_func(base_dir, repo)) for repo in repos]
+
+
+def _select_repo_path_function(
+    layout: DirectoryLayout,
+) -> Callable[[pathlib.Path, plug.StudentRepo], pathlib.Path]:
+    if layout == DirectoryLayout.FLAT:
+        return _flat_repo_path
+    elif layout == DirectoryLayout.NESTED_BY_TEAM:
+        return _nested_team_repo_path
```

Comment:
I had always known Python could return functions (since i get an error for doing that when i miss the parentheses) and used some way like this. I really like doing it this as i think its still easy to read and very clear what it does, It's also great if were going to add a more advanced function that does some calculations to get the path

With that said; Is that really going to happen? And even if it does, it doesnt need its own function, so this really only makes the code more readable, so my question is if its really necessary instead of having a single function with an if else statement that directly returns the path? Maybe theres some other motivation that you had that im missing? Otherwise it looks great, but maybe overkill.

Reply:
It's not really about how advanced the functions are, it's simply that it looks nice and is easy to read. With that said, I'm going to hide this away in a different manner as I've gotten another idea.